corehq/apps/reports/views.py

```diff
@@ -165,7 +165,16 @@ def saved_reports(request, domain, template="reports/reports_home.html"):
             or request.couch_user.get_viewable_reports()):
         raise Http404
 
-    configs = ReportConfig.by_domain_and_owner(domain, user._id)
+    all_configs = ReportConfig.by_domain_and_owner(domain, user._id)
+    good_configs = []
+    for config in all_configs:
+        if config.is_configurable_report:
+            try:
+                config.configurable_report.spec
+            # todo: there's got to be a better thing to raise in this situation
```

Comment:
I kind of agree w/ this comment.  It's a bit confusing that this is filtering out deleted reports, but if this is the best way you found I'm ok w/ it/